meilisearch-types/src/keys.rs

```diff
@@ -179,114 +180,101 @@ fn parse_expiration_date(
+    const IndexesAdd = 7;
+    const IndexesGet = 8;
+    const IndexesUpdate = 9;
+    const IndexesDelete = 10;
+    const IndexesSwap = 11;
+    const TasksAll = 12;
+    const TasksCancel = 13;
+    const TasksDelete = 14;
+    const TasksGet = 15;
+    const SettingsAll = 16;
+    const SettingsGet = 17;
+    const SettingsUpdate = 18;
+    const StatsAll = 19;
+    const StatsGet = 20;
+    const MetricsAll = 21;
+    const MetricsGet = 22;
+    const DumpsAll = 23;
+    const DumpsCreate = 24;
+    const SnapshotsAll = 25;
+    const SnapshotsCreate = 26;
+    const Version = 27;
+    const KeysAdd = 28;
+    const KeysGet = 29;
+    const KeysUpdate = 30;
+    const KeysDelete = 31;
+    const ExperimentalFeaturesGet = 32;
+    const ExperimentalFeaturesUpdate = 33;
+    }
 }
 
 impl Action {
+    // @TODO: Consider using https://github.com/rust-phf/rust-phf
+    const SERDE_MAP_ARR: [(&'static str, Self); 34] = [
+        ("*", Self::All),
+        ("search", Self::Search),
+        ("documents.*", Self::DocumentsAll),
+        ("documents.add", Self::DocumentsAdd),
+        ("documents.get", Self::DocumentsGet),
+        ("documents.delete", Self::DocumentsDelete),
+        ("indexes.*", Self::IndexesAll),
+        ("indexes.create", Self::IndexesAdd),
+        ("indexes.get", Self::IndexesGet),
+        ("indexes.update", Self::IndexesUpdate),
+        ("indexes.delete", Self::IndexesDelete),
+        ("indexes.swap", Self::IndexesSwap),
+        ("tasks.*", Self::TasksAll),
+        ("tasks.cancel", Self::TasksCancel),
+        ("tasks.delete", Self::TasksDelete),
+        ("tasks.get", Self::TasksGet),
+        ("settings.*", Self::SettingsAll),
+        ("settings.get", Self::SettingsGet),
+        ("settings.update", Self::SettingsUpdate),
+        ("stats.*", Self::StatsAll),
+        ("stats.get", Self::StatsGet),
+        ("metrics.*", Self::MetricsAll),
+        ("metrics.get", Self::MetricsGet),
+        ("dumps.*", Self::DumpsAll),
+        ("dumps.create", Self::DumpsCreate),
+        ("snapshots.*", Self::SnapshotsAll),
+        ("snapshots.create", Self::SnapshotsCreate),
+        ("version", Self::Version),
+        ("keys.create", Self::KeysAdd),
+        ("keys.get", Self::KeysGet),
+        ("keys.update", Self::KeysUpdate),
+        ("keys.delete", Self::KeysDelete),
+        ("experimental.get", Self::ExperimentalFeaturesGet),
+        ("experimental.update", Self::ExperimentalFeaturesUpdate),
+    ];
+
+    fn ser_action_to_action(v: &str) -> Option<Action> {
+        Self::SERDE_MAP_ARR
+            .iter()
+            .find(|(ser_action, _)| &v == ser_action)
+            .map(|(_, action)| *action)
+    }
+
+    fn action_to_ser_action(v: &Action) -> &'static str {
+        Self::SERDE_MAP_ARR
+            .iter()
+            .find(|(_, ref action)| v == action)
+            .map(|(ser_action, _)| ser_action)
+            .expect("`action_wanted` should always have a matching serialized value")
+    }
```

Comment:
I'm not a big fan of these, how could I improve it? Maybe use https://github.com/rust-phf/rust-phf ?

Reply:
I am pretty sure that it's ok. You are comparing at max 33 values in this array. Yes, those are strings but it's fast enough.